day1/es1/LocalButton.qml

```diff
@@ -0,0 +1,31 @@
+import QtQuick 2.0
+
+Rectangle {
+    id: button
+
+    property string text: qsTr("none")
+
+    signal clicked
+
+    width: parent.width/parent.children.length
```

Comment:
In questo caso la componente Rectangle è la componente root di questa componente.

Quando usiamo `parent` sulla root di una componente, stiamo dicendo alla componente di andare a prendersi qualche valore all'esterno: è sempre preferibile considerare ciascuna componente (nel nostro caso `LocalButton`) come una componente a sè stante, che ignora totalmente qualunque cosa ci sia all'esterno.
In generale quindi, usare `parent` sulla root di una componente viola questa norma.

Direi che per rimuovere l'uso di `parent` in binding sulle property `width` e `height` abbiamo due diverse opzioni:
- le impostiamo dall'esterno, come abbiamo fatto negli altri casi
- usiamo dei valori di default assoluti (per esempio `width: 80`) all'interno di LocalButton che siano sensati per il tipo di componente che stiamo scrivendo, valori che eventualmente possono sempre essere sovrascritti dall'esterno qualora serva maggiore personalizzazione

Direi che è indifferente quale due scegliere tra queste! :)

Reply:
rimossi tutti i parent che facevano riferimento a componenti esterni

Reply:
Bene!
A questo punto, visto che abbiamo di default `width` e `height` già corrette per la componente `LocalButton`, non ci serve in realtà sovrascriverle nuovamente dentro `SwitchButton.qml`, giusto? :)
Penso che possiamo togliere tranquillamente `width: parent.width/parent.children.length` e `height: parent.height`, tanto alla fine i valori che impostano sono identici a quelli di default! :)

Reply:
Poi ieri prima di leggere questo tuo ultimo commento avevo eliminato il settaggio con valori di default di width e height dentro LocalButton, avevo lasciato il settaggio da esterno, in questo modo si adattano automaticamente al loro “contenitore”, dimmi se può andar bene

Reply:
Come soluzione è molto generica (per l'uso di children), nel senso che già sappiamo che saranno solo 2, comunque va bene, sisi! :+1: 

Ti lascio un'informazione aggiuntiva che talvolta risulta utile: la `width` della componente Row, se non viene bindata ad un particolare valore (come in questo caso), si adatta automaticamente alla somma delle `width` delle sue componenti figlie.

Ad esempio in questo esempio qui sotto, la `width` della componente Row diventa automaticamente uguale a `225` 
```
Row {
    Rectangle {
        width: 100
    }
    Rectangle {
        width: 125
    }
}
```

Reply:
Segna pure come risolto! :)